File: entities/youtube.ts
import { z } from "zod";

export const GetYTVideoInfoFromDBSchema = z.object({
  video_id: z.string(),
  title: z.string(),
  duration: z.number(),
  thumbnail: z.string(),
  description: z.string(),
  is_embedable: z.boolean(),
  tags: z.array(z.string()),
  language: z.string(),
  category: z.string(),
  is_public: z.boolean(),
  download_status: z.string(),
});

export const PostYTVideoDownload = z.object({
  status: z.string(),
});

File: server/server/entities/youtube.py
from typing import Optional

from pydantic import BaseModel


class YoutubeUpsertSchema(BaseModel):
    user_id: str
    video_id: str
    title: Optional[str]
    duration: Optional[int]
    thumbnail: Optional[str]
    description: Optional[str]
    is_embedable: Optional[bool]
    tags: Optional[str]
    language: Optional[str]
    category: Optional[str]
    is_public: Optional[bool]
    extension: Optional[str]
    download_status: Optional[str]
    num_speakers: Optional[int]
    transcribing_status: Optional[str]


class PostYoutubeDownloadRequestSchema(BaseModel):
    user_id: str
    video_id: str
